network/p2p/scoring/internal/subscriptionCache.go

```diff
@@ -0,0 +1,154 @@
+package internal
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/rs/zerolog"
+	"go.uber.org/atomic"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module"
+	herocache "github.com/onflow/flow-go/module/mempool/herocache/backdata"
+	"github.com/onflow/flow-go/module/mempool/herocache/backdata/heropool"
+	"github.com/onflow/flow-go/module/mempool/stdmap"
+)
+
+var ErrTopicRecordNotFound = fmt.Errorf("topic record not found")
+
+type SubscriptionRecordCache struct {
+	c            *stdmap.Backend
+	currentCycle atomic.Uint64
```

Comment:
Can you add some documentation describing the relationship between the `currentCycle` and how it is moved during each updateTopics time interval which is configured. It's sort of implied here but a short concise description would be good.

Reply:
https://github.com/onflow/flow-go/pull/4988/commits/ddfcc9fe969e590b3f445188a3850960700b6e62